.github/workflows/release.yml

```diff
@@ -57,34 +83,33 @@ jobs:
             os_type: windows-x86_64
       fail-fast: false
     runs-on: ${{ matrix.os }}
+    needs: generate-tag
+
     steps:
-      - name: Clean up and validate ref to build and create artifact name
+      - name: Create artifact name
+        id: create-artifact
         shell: bash
         run: |
           # Manual dispatch on a branch ref
           if [ "${{ github.event_name == 'workflow_dispatch' }}" == "true" ]; then
-            echo "iota_ref=${{ inputs.iota_branch }}" >> $GITHUB_ENV
-            echo "artifact_name=iota-$(echo ${{ inputs.iota_branch }} | tr "/" "-")-${{ matrix.os_type }}" >> $GITHUB_ENV
+            echo "artifact_name=iota-$(echo ${{ inputs.iota_branch }} | tr "/" "-")-${{ matrix.os_type }}" >> $GITHUB_OUTPUT
           fi
 
           # Release builds on a tag ref
           if [ "${{ github.event_name == 'release' }}" == "true" ]; then
-            export iota_ref=$(echo ${{ github.ref }} | sed s/'refs\/tags\/'// )
-
-            echo "iota_ref=${iota_ref}" >> $GITHUB_ENV
-            echo "artifact_name=iota-${iota_ref}-${{ matrix.os_type }}" >> $GITHUB_ENV
+            echo "artifact_name=iota-${iota_ref}-${{ matrix.os_type }}" >> $GITHUB_OUTPUT
```

Comment:
this should be using `${{ needs.generate-tag.outputs.iota_ref }}` instead of `${iota_ref}`

Reply:
good call, missed that